### PR TITLE
Refactor Win32.Registry tests

### PR DESCRIPTION
**Pull request by Maxwe11**
- rename tests
- replace manual comparing with Assert.Equal
- share test input data
- reduce amount of fields in test classes
- removed redundant qualifiers and unused namespace usings
- const/readonly added where possible

I want to receive feedback before starting doing the same for `RegistryKey`

### CODE REVIEW

**Comment by Maxwe11** [comment]
failed due to #1205

**Comment by weshaggard** [comment]
@dotnet-bot test this please

**Comment by pallavit** [comment]
@Maxwe11 Thanks for fixing these. I left some comments. Other than that changes look good, I also tested them locally.

**Comment by Maxwe11** [comment]
@pallavit feedback rebased, build failed due to unrelated with PR reasons

File: src/Microsoft.Win32.Registry/tests/Registry/Registry_GetValue_str_str_obj.cs [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+                    key.DeleteValue(valueName);
+                }
+            }
+            catch (UnauthorizedAccessException) { }
+            catch (IOException) { }
+        }
+
+        public void Dispose()
+        {
+            _testRegistryKey.Dispose();
```
Comment (Maxwe11) [inline src/Microsoft.Win32.Registry/tests/Registry/Registry_GetValue_str_str_obj.cs]: added Dispose call here and in classes below

File: src/Microsoft.Win32.Registry/tests/Registry/Registry_GetValue_str_str_obj.cs [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+                    new object[] { Registry.Users,           "Test4" },
+                    new object[] { Registry.PerformanceData, "Test5" },
+                    new object[] { Registry.CurrentConfig,   "Test6" },
+                };
+            }
+        }
+
+        [Theory]
+        [MemberData("TestRegistryKeys")]
+        public static void GetValueFromDifferentKeys(RegistryKey key, string valueName)
```
Comment (Maxwe11) [inline src/Microsoft.Win32.Registry/tests/Registry/Registry_GetValue_str_str_obj.cs]: marked with `static`

**Comment by Maxwe11** [comment]
@dotnet-bot test this please

**Comment by Maxwe11** [comment]
@pallavit sorry for the delay, rebased. Also take a look #1615

**Comment by pallavit** [comment]
Thanks for cleaning this up. I am taking a look at the changes. Adding @tarekgh for another set of eyes on these.

File: src/Microsoft.Win32.Registry/tests/Registry/Registry_Getvalue_str_str_obj.cs [hunk 3]
```diff
@@ -1,304 +1,109 @@
...
-                    Assert.False(true, "ERROR: Incorrect value read. Expected: " + newValue + " Actual: " + Microsoft.Win32.Registry.GetValue("HKEY_CURRENT_USER", "Test1", defaultValue));
-                }
-                Microsoft.Win32.Registry.CurrentUser.DeleteValue("Test1");
-            }
-            catch (Exception exc)
-            {
-                Assert.False(true, "ERROR: unexpected exception, " + exc.ToString());
-            }
+            _testRegistryKey.SetValue(valueName, testValue);
+            Assert.Equal(testValue.ToString(), Registry.GetValue(_testRegistryKey.Name, valueName, null).ToString());
```
Comment (tarekgh) [inline src/Microsoft.Win32.Registry/tests/Registry/Registry_Getvalue_str_str_obj.cs]: Should we wrap this line under try-finally to make sure we'll delete the key?

Comment (Maxwe11) [inline src/Microsoft.Win32.Registry/tests/Registry/Registry_Getvalue_str_str_obj.cs]: not necessary, value will be deleted with key in Dispose

File: src/Microsoft.Win32.Registry/tests/RegistryKey/RegistryKey_DeleteSubKey_Str_Bln.cs [hunk 4]
```diff
@@ -1,148 +1,80 @@
...
+            const string name = "Test";
 
-        [Fact]
-        public void Test02()
-        {
-            Action a = () => { _rk1.DeleteSubKey(null, true); };
-            Assert.Throws<ArgumentNullException>(() => { a(); });
-        }
+            // Should throw if passed subkey name is null
+            Assert.Throws<ArgumentNullException>(() => _testRegistryKey.DeleteSubKey(null, true));
```
Comment (pallavit) [inline src/Microsoft.Win32.Registry/tests/RegistryKey/RegistryKey_DeleteSubKey_Str_Bln.cs]: Should we not also add (null, false) as was done earlier.

File: src/Microsoft.Win32.Registry/tests/Registry/Registry_Getvalue_str_str_obj.cs (hunk 3 above)
Comment (tarekgh) [inline src/Microsoft.Win32.Registry/tests/Registry/Registry_Getvalue_str_str_obj.cs]: Sorry I meant to cleanup the value.

**Comment by tarekgh** [comment]
LGTM. Thanks @Maxwe11 for cleaning up all of these tests. great job :-)

File: src/Microsoft.Win32.Registry/tests/RegistryKey/RegistryKey_DeleteSubKey_Str_Bln.cs (hunk 4 above)
Comment (Maxwe11) [inline src/Microsoft.Win32.Registry/tests/RegistryKey/RegistryKey_DeleteSubKey_Str_Bln.cs]: will fix

**Comment by Maxwe11** [comment]
Consider: could we enable parallel test running with these changes?

**Comment by tarekgh** [comment]
@Maxwe11 are we sure there is not 2 test cases using the same registry key name? if so we may get a issue with parallel running

**Comment by Maxwe11** [comment]
>are we sure there is not 2 test cases using the same registry key name?

Seems to me yes. Each test collection has use unique subkey name.

**Comment by pallavit** [comment]
LGTM.

**Comment by tarekgh** [comment]
@Maxwe11 we may try first without enabling parallel runs and then we have confidence the tests are stable we can enable parallel runs.

**Comment by Maxwe11** [comment]
@pallavit feedback addressed

**Comment by pallavit** [comment]
LGTM. @Maxwe11 would you want to squash these in a single merge? If not, I will merge this PR as-is.

**Comment by Maxwe11** [comment]
feedback squashed

**Comment by pallavit** [comment]
Thanks. I will merge the PR once the CI completes.

**Comment by Maxwe11** [comment]
CI green

**Comment by pallavit** [comment]
@Maxwe11 Thanks again for taking a look at this. Merging these changes.

**Pull request merged**